pandas/core/algorithms.py

```diff
@@ -2149,3 +2149,23 @@ def _sort_tuples(values: np.ndarray[tuple]):
     arrays, _ = to_arrays(values, None)
     indexer = lexsort_indexer(arrays, orders=True)
     return values[indexer]
+
+
+def make_duplicates_of_left_unique_in_right(
```

Comment:
is this related to or useful for the index.union-with-duplicates stuff?

Reply:
If you pass in the union as left and right, you would get the distinct result. Have to take a look if we can use this.